pkg/controller/jobs/appwrapper/appwrapper_controller_test.go

```diff
@@ -132,6 +133,7 @@ func TestPodSets(t *testing.T) {
 
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
+			features.SetFeatureGateDuringTest(t, features.TopologyAwareScheduling, true)
```

Comment:
IIUC the tests would pass even without changing the code. Let's introduce a new param enableTopologyAwareScheduling, and add a sanity test when `enableTopologyAwareScheduling=false`.

Reply:
I've added these changes only to the tests that were failing, so enabling the feature gate is required. Will add a `enableTopologyAwareScheduling=false` case

Reply:
Added `enableTopologyAwareScheduling` to the affected tests, as well as sanity tests where it is intentionally disabled.